spyder/widgets/fileswitcher.py

```diff
@@ -287,44 +290,46 @@ def __init__(self, parent, tabs, data):
         self.list.itemSelectionChanged.connect(self.item_selection_changed)
         self.list.clicked.connect(self.edit.setFocus)
 
-        # Setup
-        self.save_initial_state()
-        self.set_dialog_position()
-        self.setup()
-
     # --- Properties
     @property
-    def editors(self):
-        return [self.tabs.widget(index) for index in range(self.tabs.count())]
+    def widgets(self):
+        return [(tabs.widget(index), plugin) for tabs,
+                plugin in self.plugins_tabs for index in range(tabs.count())]
 
     @property
     def line_count(self):
-        return [editor.get_line_count() for editor in self.editors]
+        return [widget.get_line_count() for widget in self.widgets]
 
     @property
     def save_status(self):
-        return [getattr(td, 'newly_created', False) for td in self.data]
+        return [getattr(td,
+                        'newly_created',
+                        False) for da in self.plugins_data for td in da]
```

Comment:
This is too complex to have it as list comprehension. Please simplify it using two list comprehensions instead.